app/scripts/controllers/account-tracker-controller.ts

```diff
@@ -113,52 +196,48 @@ export default class AccountTracker {
 
   #currentBlockNumberByChainId: Record<Hex, string | null> = {};
 
-  #getCurrentChainId: AccountTrackerOptions['getCurrentChainId'];
-
-  #getNetworkIdentifier: AccountTrackerOptions['getNetworkIdentifier'];
-
-  #preferencesController: AccountTrackerOptions['preferencesController'];
-
-  #onboardingController: AccountTrackerOptions['onboardingController'];
+  #getNetworkIdentifier: AccountTrackerControllerOptions['getNetworkIdentifier'];
```

Comment:
Now that extension is using core's `NetworkController`, is it necessary to preserve the logic for pulling from the redux store using `getProviderConfig`? Is there anything keeping us from replacing this with a `NetworkController:getState` call that extracts `selectedNetworkClientId`?

Reply:
Now that we know we'll be migrating this to the core version that extends `StaticIntervalPollingController` we definitely will need to figure out a way to remove this line. 

I don't think we'd ever want a core controller relying on the redux store for stateful data. Seems like it would potentially introduce side effects and/or be a circular dependency?

Since the scope of this PR keeps the extension controller in use I guess we can handle this in the migration PR.

Reply:
Yes, this will be handled in the migration PR. I’ll verify if it’s still needed, as it’s not part of the core implementation and is currently only being used to interact with the RPC using localhost as the hostname.

Reply:
Plugging in the core controller into where the extension controller used to be will have the effect of removing this callback and its functionality. We should def keep an eye out in case it breaks anything, but it sounds like you have a good handle on what it's doing. :)